backend-lib/src/main/rust/lib.rs

```diff
@@ -2054,6 +2056,76 @@ pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_createPcz
     unwrap_exc_or(&mut env, res, ptr::null_mut())
 }
 
+/// Redacts information from the given PCZT that is unnecessary for the Signer role.
+///
+/// Returns the updated PCZT in its serialized format.
+#[unsafe(no_mangle)]
+pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_redactPcztForSignerRole<
+    'local,
+>(
+    mut env: JNIEnv<'local>,
+    _: JClass<'local>,
+    pczt: JByteArray<'local>,
+) -> jbyteArray {
+    let res = catch_unwind(&mut env, |env| {
+        let _span = tracing::info_span!("RustBackend.redactPcztForSignerRole").entered();
+
+        let pczt = Pczt::parse(&env.convert_byte_array(pczt)?[..])
+            .map_err(|e| anyhow!("Invalid PCZT: {:?}", e))?;
+
+        let pczt_with_proofs = Redactor::new(pczt)
+            .redact_global_with(|mut r| r.redact_proprietary("zcash_client_backend:proposal_info"))
+            .redact_orchard_with(|mut r| {
+                r.redact_actions(|mut ar| {
+                    ar.clear_spend_witness();
+                    ar.redact_output_proprietary("zcash_client_backend:output_info");
+                })
+            })
+            .redact_sapling_with(|mut r| {
+                r.redact_spends(|mut sr| sr.clear_witness());
+                r.redact_outputs(|mut or| {
+                    or.redact_proprietary("zcash_client_backend:output_info")
+                });
+            })
+            .redact_transparent_with(|mut r| {
+                r.redact_outputs(|mut or| {
+                    or.redact_proprietary("zcash_client_backend:output_info")
+                });
+            })
+            .finish();
```

Comment:
This is effectively a blacklist of proprietary data. Can it be a whitelist (which might contain nothing, I'm not sure)?

Reply:
I don't think so. I originally considered just clearing all proprietary data, but the PCZT hasn't necessarily only been generated locally and there e.g. might be FROST proprietary data (in the future where Keystone supports FROST). So I decided on only removing things that we knew we'd not need on the Keystone device (and that we could recover because we were holding onto proofs).

In the Keystone firmware, by comparison, we'll be clearing everything we don't need for the return trip because there we know the scanner has it all.